src/engine/Metavariable_pattern.ml

```diff
@@ -122,13 +122,20 @@ let get_nested_metavar_pattern_bindings get_nested_formula_matches env r mvar
       (* We will create a temporary file with the content of the metavariable,
        * then call evaluate_formula recursively. *)
       match MV.range_of_mvalue mval with
-      | None ->
-          error env
-            (Common.spf
-               "metavariable-pattern failed because we lack range info for %s, \
-                please file a bug report"
-               mvar);
-          []
+      | None -> (
+          match mval with
+          | MV.Ss [] ->
```

Comment:
there are actually more cases like this. See Core_json_output.range_of_any_opt.
Maybe there is some code factorization to do, maybe a helper function in Metavariable.ml

Also would be good to add a test showing that now we don't crash on the code you've added.

Reply:
i can't add a test i think, because the "crash" isn't really a crash, its just a reported error message

Reply:
(i could add a CLI test, but i was thinking because pysemgrep is going away, that may not be worth it)